internal/services/arckubernetes/arc_kubernetes_cluster_extension_resource.go

```diff
@@ -0,0 +1,358 @@
+				Type:         pluginsdk.TypeString,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"configuration_settings": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"release_train": {
+			Type:          pluginsdk.TypeString,
+			Optional:      true,
+			Computed:      true,
+			ForceNew:      true,
+			ConflictsWith: []string{"version"},
+			ValidateFunc:  validation.StringIsNotEmpty,
+		},
+
+		"release_namespace": {
+			Type:          pluginsdk.TypeString,
+			Optional:      true,
+			Computed:      true,
+			ForceNew:      true,
+			ConflictsWith: []string{"target_namespace"},
+			ValidateFunc:  validation.StringIsNotEmpty,
+		},
+
+		"target_namespace": {
+			Type:          pluginsdk.TypeString,
+			Optional:      true,
+			Computed:      true,
+			ForceNew:      true,
+			ConflictsWith: []string{"release_namespace"},
+			ValidateFunc:  validation.StringIsNotEmpty,
+		},
+
+		"version": {
+			Type:          pluginsdk.TypeString,
+			Optional:      true,
+			ForceNew:      true,
+			ConflictsWith: []string{"release_train"},
+			ValidateFunc:  validation.StringIsNotEmpty,
+		},
+	}
+
+}
+
+func (r ArcKubernetesClusterExtensionResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"current_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r ArcKubernetesClusterExtensionResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ArcKubernetesClusterExtensionModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.ArcKubernetes.ExtensionsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			clusterID, err := arckubernetes.ParseConnectedClusterID(model.ClusterID)
+			if err != nil {
+				return err
+			}
+
+			// defined as strings because they're not enums in the swagger https://github.com/Azure/azure-rest-api-specs/pull/23545
+			id := extensions.NewExtensionID(subscriptionId, clusterID.ResourceGroupName, "Microsoft.Kubernetes", "connectedClusters", clusterID.ConnectedClusterName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			autoUpgradeMinorVersion := false
+			if model.Version == "" {
+				autoUpgradeMinorVersion = true
+			}
+
+			properties := &extensions.Extension{
+				Properties: &extensions.ExtensionProperties{
+					AutoUpgradeMinorVersion:        &autoUpgradeMinorVersion,
```

Comment:
Should this property be exposed as a bool e.g. `auto_upgrade_enabled` instead of string for version - it's misleading that omitting a value or version will auto upgrade the extension version

Reply:
I think the answer is no. The value of `AutoUpgradeMinorVersion` totally depends on the value of `version`. If `version` is set, `AutoUpgradeMinorVersion` is disabled by Azure Service, and vice-versa, which means if you set a value for `version` and set `auto_upgrade_enabled = true`, the Azure Service will change it back to `false`. We cannot use `auto_upgrade_enabled` instead of string for `version` because `version` is not a fixed value. It may point to any valid version number, like`1.2.0`, `2.3.0`, etc.